flume-ng-channels/flume-kafka-channel/src/main/java/org/apache/flume/channel/kafka/KafkaChannel.java

```
@@ -564,14 +564,22 @@ protected void doCommit() throws InterruptedException {
                   ex);
         }
       } else {
+        //event taken ensures that we have collected events in this transaction
+        // before committing
         if (consumerAndRecords.get().failedEvents.isEmpty() && eventTaken) {
```

Naive question: could you please tell why the else path is not needed. Condition to the if statement suggests that we silently ignore some failure scenario.

---

See comment